src/tz_ohos.rs

```diff
@@ -0,0 +1,36 @@
+//! OpenHarmony doesn't have `/etc/localtime`, we have to use it's "Time Service" to get the timezone information:
+//!
+//! - [API Refence](https://gitee.com/openharmony/docs/blob/43726785b4033887cd1a838aaaca5e255897a71e/en/application-dev/reference/apis-basic-services-kit/time__service_8h.md)
+
+use crate::GetTimezoneError;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+#[repr(C)]
+#[allow(non_camel_case_types)]
+#[allow(dead_code)]
+enum TimeService_ErrCode {
+    TIMESERVICE_ERR_OK = 0,
+    TIMESERVICE_ERR_INTERNAL_ERROR = 13000001,
+    TIMESERVICE_ERR_INVALID_PARAMETER = 13000002,
+}
+
+#[link(name = "time_service_ndk", kind = "dylib")]
+extern "C" {
+    fn OH_TimeService_GetTimeZone(timeZone: *mut std::ffi::c_char, len: u32)
+        -> TimeService_ErrCode;
+}
+
+pub(crate) fn get_timezone_inner() -> Result<String, GetTimezoneError> {
+    let mut time_zone = [0; 40];
+    let ret = unsafe { OH_TimeService_GetTimeZone(time_zone.as_mut_ptr(), time_zone.len() as u32) };
```

Comment:
Please add a `// SAFETY:` comment for the `unsafe` as described [here](https://rust-lang.github.io/rust-clippy/stable/index.html#undocumented_unsafe_blocks).

Also, the docs are not perfectly clear to me: the `len` parameter is "without maximum limit", so should it actually be `time_zon.len()-1`? This is also the understanding I get when I read the error description for `TIMESERVICE_ERR_INVALID_PARAMETER `: "The time zone is null or the length of the time zone name (excluding the end character \0) is greater than or equal to len." It seems like including the `-1` will make the code safer in that it specifies the `time_zone` array will end with `\0`.

Reply:
Yes, have to admit the `-1` is needed from the doc. But the [official test](https://github.com/openharmony/time_time_service/blob/93ada192086efb7caf4033e6896e8fff3cd22943/test/unittest/native_test/time_service_native_test.cpp#L37) does use `time_zone.len()` directly, which is misleading. Anyway, using `len - 1` makes it safer.